Echo/PolySyncGetOpt.hpp

```diff
@@ -0,0 +1,111 @@
+/**
+ * @file PolySyncGetOpt.hpp
+ * @brief PolySync GetOpt Command Line Utility.
+ *
+ * PUBLIC_HEADER
+ */
+
+#ifndef POLYSYNCGETOPT_HPP
+#define POLYSYNCGETOPT_HPP
+
+#include <PolySyncDataModel.hpp>
+#include <getopt.h>
+
+/**
+ * @namespace polysync
+ */
+namespace polysync
+{
+
+/**
+ * @brief PolySyncGetOpt class
+ *
+ * The PolySyncGetOpt C++ class exists to build upon the C-based getopt class.
+ * Getopt parses command line input, and validates that input with options,
+ * arguments, and option-arguments.
+ */
+class PolySyncGetOpt
+{
+
+public:
+
+    /**
+     * @brief Determines whether user supplied cmd line options are supported.
+     * @param optchar Character of user supplied option on cmd line.
+     * @return Returns -1 if not supported; returns index if supported (O-N)
+     */
+    int getOptIdx( const char optret );
+
+    /**
+     * @brief Parses cmd line arguments in C getopt style.
+     * @param argv Argument vector.
+     * @param argc Argument count.
+     * @return
+     */
+    bool optionsParse( const int argc, char * argv[] );
+
+    /**
+     * @brief Member variable getter for message(s) name(s).
+     * @return Returns char * variable containing message name.
+     */
```

Comment:
1.) Move these function definitions to the source file. 

2.) Always use `std::string` for strings.

3.) Getters should be `const`.  ---->   std::string getMessageName() const;